RFS/src/main/java/org/opensearch/migrations/bulkload/workcoordination/OpenSearchWorkCoordinator.java

```diff
@@ -935,7 +1031,13 @@ private void refresh(Supplier<IWorkCoordinationContexts.IRefreshContext> context
                     switch (obtainResult) {
                         case SUCCESSFUL_ACQUISITION:
                             ctx.recordAssigned();
-                            return getAssignedWorkItem(leaseChecker, ctx);
+                            var workItem = getAssignedWorkItem(leaseChecker, ctx);
+                            if (workItem.successorWorkItemIds != null) {
+                                // continue the previous work of creating the successors and marking this item as completed
+                                createSuccessorWorkItemsAndMarkComplete(workItem.workItemId, workItem.successorWorkItemIds, ctx::getCreateSuccessorWorkItemsContext);
+                                return new AlreadyCompleted();
```

Comment:
What's the downside to continuing?  AlreadyCompleted is a weird response for acquireNextWorkItem().  Why would it have been acquired if it was already done?